src/execution/operator/aggregate/physical_ungrouped_aggregate.cpp

```diff
@@ -22,6 +22,21 @@ PhysicalUngroupedAggregate::PhysicalUngroupedAggregate(vector<LogicalType> types
                                                        idx_t estimated_cardinality)
     : PhysicalOperator(PhysicalOperatorType::UNGROUPED_AGGREGATE, move(types), estimated_cardinality),
       aggregates(move(expressions)) {
+
+	vector<idx_t> distinct_indices;
+	for (idx_t i = 0; i < aggregates.size(); i++) {
+		auto &aggregate = aggregates[i];
+		auto &aggr = (BoundAggregateExpression &)*aggregate;
+		if (aggr.IsDistinct()) {
+			distinct_indices.push_back(i);
+			distinct_filter.push_back(i);
+		}
+	}
+	if (distinct_indices.empty()) {
+		return;
+	}
+	distinct_collection_info = make_unique<DistinctAggregateCollectionInfo>(aggregates, move(distinct_indices));
```

Comment:
This construction looks quite similar here and in the `physical_hash_aggregate`, could we unify this?